src/Uno.UI/UI/Xaml/Controls/Image/Image.Android.cs

```diff
@@ -242,6 +245,40 @@ private bool MustOpenImageToMeasure()
 			return (Stretch == Stretch.Uniform || Stretch == Stretch.None) && (double.IsNaN(Width) || double.IsNaN(Height));
 		}
 
+		private void SetSourceUriFromFile(ImageSource newImageSource)
```

Comment:
This looks like a duplicate from the method below. What's different ?

Reply:
It'd be best not to duplicate the method, but adjust its content.

Reply:
The old solution was using FireForget Pattern. There was no guarantee that the Task will be completed. If any developer put task.delay on buttonclick will worked in 100% because the task is completed.

We created a synchronous path to wait the bitmapdecoder create a memory representation of the image